src/test/java/org/apache/commons/text/similarity/JaccardDistanceTest.java

```diff
@@ -36,21 +36,23 @@ public static void setUp() {
 
     @Test
     public void testGettingJaccardDistance() {
-        assertEquals(1.00d, classBeingTested.apply("", ""), 0.00000000000000000001d);
-        assertEquals(1.00d, classBeingTested.apply("left", ""), 0.00000000000000000001d);
-        assertEquals(1.00d, classBeingTested.apply("", "right"), 0.00000000000000000001d);
-        assertEquals(0.25d, classBeingTested.apply("frog", "fog"), 0.00000000000000000001d);
-        assertEquals(1.00d, classBeingTested.apply("fly", "ant"), 0.00000000000000000001d);
-        assertEquals(0.78d, classBeingTested.apply("elephant", "hippo"), 0.00000000000000000001d);
-        assertEquals(0.36d, classBeingTested.apply("ABC Corporation", "ABC Corp"), 0.00000000000000000001d);
-        assertEquals(0.24d, classBeingTested.apply("D N H Enterprises Inc", "D & H Enterprises, Inc."),
-                0.00000000000000000001d);
-        assertEquals(0.11d, classBeingTested.apply("My Gym Children's Fitness Center", "My Gym. Childrens Fitness"),
-                0.00000000000000000001d);
-        assertEquals(0.10d, classBeingTested.apply("PENNSYLVANIA", "PENNCISYLVNIA"), 0.00000000000000000001d);
-        assertEquals(0.87d, classBeingTested.apply("left", "right"), 0.00000000000000000001d);
-        assertEquals(0.87d, classBeingTested.apply("leettteft", "ritttght"), 0.00000000000000000001d);
-        assertEquals(0.0d, classBeingTested.apply("the same string", "the same string"), 0.00000000000000000001d);
+        // Results generated using the python distance library using:
+        // distance.jaccard(seq1, seq2)
+        assertEquals(1.0, classBeingTested.apply("", ""));
+        assertEquals(1.0, classBeingTested.apply("left", ""));
+        assertEquals(1.0, classBeingTested.apply("", "right"));
+        assertEquals(0.25, classBeingTested.apply("frog", "fog"));
+        assertEquals(1.0, classBeingTested.apply("fly", "ant"));
+        assertEquals(0.7777777777777778, classBeingTested.apply("elephant", "hippo"));
```

Comment:
Is it safe to rely on the precision here? Or should we use something like (not tested) `assertEquals(0.7777777777777778, classBeingTested.apply("elephant", "hippo"), 0.1);` ? Where 0.1 is a value for more/less precision loss.

Reply:
I understand where you are coming from. However it currently passes the test because the underlying math is using `intersect / divide` with integers cast to double. So the value can be exact and there is not really a way to do it with less precision unless the algorithm is wrong.

I can update to using an epsilon of 1e-10 for all values that are not expected to be 0.0 or 1.0. However that does make it possible someone reading the test will think the output value is subject to imprecision because it is not.

How about a comment in the test explaining where each value comes from, or even the actual computation:

```
assertEquals(0.7777777777777778, classBeingTested.apply("elephant", "hippo"));
// becomes
assertEquals(7.0 / 9.0, classBeingTested.apply("elephant", "hippo"));
```

Reply:
>How about a comment in the test explaining where each value comes from, or even the actual computation

I will take your word here :-) let's leave it like this for now then. If it eventually fails in a JVM, we can either add that episillon where appropriate, or the comment, or the explicit calculation (liked this last one, never occurred me to test that way!).

:+1: from me. And travis is happy too. Up to you to merge it now or wait for others to review :)

Reply:
I will redo the test with the explicit computation. The numbers taken from the Python library can appear arbitrary. Including the computation is a far more explicit test.